filecoin-proofs/src/api/internal.rs

```diff
@@ -474,9 +473,15 @@ pub fn seal<T: Into<PathBuf> + AsRef<Path>>(
         &replica_id,
         &mut data,
         None,
-    )?;
+    )
+    .map_err(|e| {
```

Comment:
The file with path `out_file` will leak under the following circumstances:

- `OpenOptions::new().read(true).write(true).open(&out_path)` is `Err`
- `f_data.set_len(sector_bytes as u64)` is `Err`
- `MmapOptions::new().map_mut(&f_data)` is `Err`

I suggest that you handle each of those cases as you do here - by passing a cleanup function to `map_err`.

Reply:
@laser Good catch. I pushed a change to handle this right. We could refine that approach to be slicker in the future, but I think what I did (add a wrapper around the path, implementing `Drop`) at least provides the semantics we want. Let me know if that works for you.